language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -147,25 +146,25 @@ lookupNameIndex = lookupUnique . coerce . view unique
 
 -- | An 'IdentifierState' includes a map indexed by 'Int's as well as a map
 -- indexed by 'ByteString's. It is used during parsing.
-type IdentifierState = (UniqueMap Unique BSL.ByteString, M.Map BSL.ByteString Unique, Unique)
+type IdentifierState = (M.Map BSL.ByteString Unique, Unique)
```

Comment:
While you're at it, why not get rid of the second bit too and just use `MonadQuote` in `newIdentifier`? I think that will simplify quite a bit.